src/ComunicaEngine.ts

```diff
@@ -1,24 +1,69 @@
 import { newEngine } from '@comunica/actor-init-sparql-solid';
+import type { ActorInitSparql } from '@comunica/actor-init-sparql';
+import type { DataSources, IDataSource } from '@comunica/bus-rdf-resolve-quad-pattern';
+import type * as RDF from '@rdfjs/types';
+import type { BindingsStream, Bindings } from '@comunica/types';
+
+export type DataSource = URL | RDF.NamedNode | IDataSource;
+
+// Change over to this type once https://github.com/microsoft/TypeScript/issues/47208#issuecomment-1014128087 is resolved
+// export type RawDataSources = DataSource | Promise<RawDataSources> | RawDataSources[];
+export type MaybePromiseArray<T> = T | T[] | Promise<T>;
+export type NestPromiseArray<T> = MaybePromiseArray<MaybePromiseArray<MaybePromiseArray<MaybePromiseArray<MaybePromiseArray<T>>>>>
+export type RawDataSources = NestPromiseArray<DataSource>;
+
+/**
+ * Settings for the ldflex-comunica engine
+ */
+export interface IEngineSettings {
+
+  /**
+   * The comunica engine to use
+   * @default actor-init-sparql-solid
+   */
+  engine?: ActorInitSparql;
+
+  /**
+   * The destination to use
+   * @default source The first source given
+   */
+  destination?: RawDataSources;
+
+  /**
+   * The context options to give to
+   * the comunica engine
+   * @default {}
+   */
+  options?: any;
+}
 
 /**
  * Asynchronous iterator wrapper for the Comunica SPARQL query engine.
  */
 export default class ComunicaEngine {
+  private _sources: Promise<DataSources>;
```

Comment:
In the future, we can drop the underscore from such private members. But there are still JavaScript consumers for now, so might want to hold off. (I use the `_` in JS to signal private, but TypeScript obsoletes that.)

Reply:
Yep - I am one such person (for instance overriding `_engine` before the custom engine option was added). I'll add a comment to change the naming in the release of the next *major* version.